packages/js/core/src/types/Uri.ts

```diff
@@ -194,4 +209,17 @@ export class Uri {
   public toJSON(): string /* $ */ {
     return this._config.uri;
   }
+
+  private static inferAuthority(_path: string): UriConfig | undefined {
+    const re = /^(?<authority>[a-z][a-z0-9-_]+):\/\/(?<path>.*)$/;
```

Comment:
Why not support full UTF-8 for authority (and path)?

Reply:
I think that's a different discussion. Let's leave changes to our URI requirements out of scope of this PR.